include/zephyr/sd/sd.h

```diff
@@ -64,6 +77,8 @@ struct sd_card {
 	enum card_status status; /*!< Card status */
 	enum card_type type; /*!< Card type */
 	uint32_t flags; /*!< Card flags */
+	uint32_t cccr_flags; /*!< SDIO CCCR data */
+	struct sdio_func func0; /*!< Function 0 common card data */
```

Comment:
```suggestion
    struct sdio_func funcs[CONFIG_SDIO_MAX_IO_FUNCS];
```
Instead of having to maintain the function objects in the application, I would prefer, that the `sd_card` structure has an array with all functions, e.g. with a configurable static size. The function would not have a `sdio_func*` as argument, but a value from the `sdio_func_num` enumeration.

Reply:
My reasoning for delegating the function objects to the application was that the SDIO driver will know how many card functions it needs to access for the SDIO device it is interfacing with. If we use something like `CONFIG_SDIO_MAX_IO_FUNCS`, then we would have a few issues:
- if multiple SDIO drivers are used, who decides the maximum number of IO functions supported?
- if one SDIO driver needs 7 functions and the other needs only 1, we have some wasted space in the SDIO structure.

Is there an issue you are running into when designing around this API that you feel would be solved by storing the SDIO functions in the card structure?

Reply:
> Is there an issue you are running into when designing around this API that you feel
> would be solved by storing the SDIO functions in the card structure?
Now I need a wrapper, which manages the SDIO function blocks. Both ways are acceptable from my point of view.

Reply:
Understood- if it is not impacting your ability to design around the API, I'd prefer we continue to delegate allocation of SDIO function blocks to the application. For what it's worth, this is similar to how Linux defines the SDIO API: https://github.com/torvalds/linux/blob/master/include/linux/mmc/sdio_func.h#L144